packages/ecs/src/topic.ts

```diff
@@ -0,0 +1,52 @@
+export type Topic<T extends string = string, D = any> = {
+  /**
+   * A property for the name of the topic
+   */
+  name: string
+
+  /**
+   * Provides iterator syntax to the consumers of a topic. This will loop
+   * through all of the events with the type specified in the second type 
+   * parameter. 
+   */
+  [Symbol.iterator](): IterableIterator<D>;
+
+  /**
+   * Pushes an event to the topic so that it can be consumed by downstream 
+   * systems.
+   */
+  pushEvent(event: D): void
+
+  /**
+   * Utility method that cleans the event list in the topic such that at the 
+   * end of the tick (or after all the systems have consumed whatever events
+   * they are interested in we can flush out events that have already been 
+   * consumed.)
+   */
+  flush(): void
+}
+
+/**
+ * The utility method used to create a topic, consumes a type parameter 
+ * and a name and returns an object that conforms to the topic type
+ */
+export const createTopic = <E>(name: string): Topic<string, E> => {
+ 
+  let events: E[] = [];
+
+  const pushEvent = (event: E) => {
+    events.push(event)
+  }
+
+  const flush = () => {
+    events = [];
+  }
+
+
+  return {
+    name,
+    [Symbol.iterator]: events.values,
```

Comment:
I just used events.values for now. The main issue is there is no way to guarantee ordering - but i don't think will be an issue unless we have lots of asynchronous processes accessing events in such a way that order matters.